sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -332,6 +334,24 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       assert(version === SPARK_VERSION_SHORT)
     }
   }
+
+  test("[SPARK-26859] Reading ORC files with explicit schema can result in wrong data") {
+    withSQLConf(SQLConf.ORC_VECTORIZED_READER_ENABLED.key -> "false") {
```

Comment:
We need to disable vectorized orc to test this bug?

Reply:
@viirya the `OrcDeserializer` class (which is the one that contains the bug) is only used for the non-vectorized reader AFAICT, so the vectorized doesn't go through the buggy codepath.

Reply:
I can't easily examine the codebase for now to check it, so thanks for explaining it.